src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -227,6 +238,9 @@ function setInitialDOMProperties(
       // Noop
     } else if (registrationNameModules.hasOwnProperty(propKey)) {
       if (nextProp) {
+        if (__DEV__ && typeof nextProp !== 'function') {
```

Comment:
That's how it used to do but I wanted to avoid extra overhead for calls for every single prop. These things are small but sometimes add up (like when we fixed DEV performance in 15.3).